enhancements/sig-architecture/105-aws-iam-registration/README.md

```diff
@@ -0,0 +1,464 @@
+```shell
+aws eks update-kubeconfig --name <hub-eks-cluster-name> --kubeconfig <path/to/create/kubeconfig> --role-arn <role-on-hub-for-managed>
+```
+
+The klusterlet/registration agent on managed cluster must assume an IAM role, in the AWS account in which managed cluster is running, on startup, with the permissions allowing it assume the role on hub IAM. The role will be tagged with managed cluster name which can be parsed from the `--cluster-name` option passed to `clusteradm join` command. The managed cluster admin must know in advance, which hub cluster to join, as the role will be tagged with the name and aws account of the hub cluster.
+
+The role name and policies created on managed cluster IAM will follow a templates to ensure standardization. The role name will contain md5 hash of the values of tags to uniquely identify.
+
+```shell
+arn:aws:iam::<managed-cluster-account-id>:role/ocm-spoke-<md5-hash-unique-identifier>
+```
+
+Another reason to follow a standard name is to ensure that the role can be created before running `clusteradm join` command and the klusterlet/registration agent can assume that role, on startup, with a standardized name.
+
+#### Graceful resource cleanup on both hub and managed cluster when managed cluster triggers unjoin
+
+The `ManagedCluster` will have a new boolean field `clientUnjoinsHub` with the default value of false. The `clusteradm unjoin` command by managed cluster admin will update the `ManagedCluster` on hub and set the `spec.clientUnjoinsHub` to true, which will trigger the new `aws-irsa` controller on hub to clean-up all the managed cluster specific resources, like IAM roles and policies on hub, and ultimately delete the `ManagedCluster`.
+
+or
+
+The unjoin command will update the `ManagedCluster` on hub and add a new condition `ManagedClusterUnjoined`, which will trigger the new `aws-irsa` controller on hub to clean-up all the managed cluster specific resources, like IAM roles and policies on hub, and ultimately delete the `ManagedCluster`.
+
+
+#### Managed cluster permission isolation on hub
+
+Every managed cluster will have a separate role on both managed cluster as well as hub IAM. Every managed cluster role on hub will allow access to managed cluster's namespace only. All roles will follow a standard naming convention, and the IAM trust policies allow roles to be assumed by relevant resources (pods, clusters, etc) only.
+
+This will ensure that roles can only be assumed by the consumers they are meant for, such that one managed cluster cannot assume roles and access resources belonging to other managed cluster
+
+### Workflow Details
+
+This is to describe the various process of cluster join
+
+Actors:
+1. cluster-admin on managed cluster
+2. AWS IAM admin on managed cluster AWS account
+3. cluster-admin on hub cluster
+4. AWS IAM admin on hub cluster AWS account
+5. hub controller
+6. agent on managed cluster
```

Comment:
the opt-in on both sides appears to work and either can revoke the connection.  Additionally, with different namespaces for the klusterlet, it appears that the same managedcluster could join multiple hubs (important for migration over time).